optaplanner-integration/optaplanner-spring-boot-example/src/main/java/org/optaplanner/spring/boot/example/service/TimetableService.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.spring.boot.example.service;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.optaplanner.core.api.solver.SolverFuture;
+import org.optaplanner.core.api.solver.SolverManager;
+import org.optaplanner.spring.boot.example.domain.Lesson;
+import org.optaplanner.spring.boot.example.domain.Room;
+import org.optaplanner.spring.boot.example.domain.TimeTable;
+import org.optaplanner.spring.boot.example.domain.Timeslot;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@Service
+public class TimetableService {
```

Comment:
rename to TimeTableService (the T of Table is capital)
TimeTableService and TimeTableController are duplicates in this _simple_ example (not in in-depth architectures), there can only be one.